sdk/ai/Azure.AI.Inference/src/Telemetry/OpenTelemetryScope.cs

```diff
@@ -0,0 +1,247 @@
+        /// <exception cref="NotSupportedException"></exception>
+        public static string GetContent(ChatRequestMessage message)
+        {
+            if (message.GetType() == typeof(ChatRequestAssistantMessage))
+                return ((ChatRequestAssistantMessage)message).Content;
+            if (message.GetType() == typeof(ChatRequestSystemMessage))
+                return ((ChatRequestSystemMessage)message).Content;
+            if (message.GetType() == typeof(ChatRequestToolMessage))
+                return ((ChatRequestToolMessage)message).Content;
+            if (message.GetType() == typeof(ChatRequestUserMessage))
+                return ((ChatRequestUserMessage)message).Content;
+            throw new NotSupportedException("The message type is not supported");
+        }
+
+        /// <summary>
+        /// Get the activity, if the telemetry is enabled. If there is no listeners for
+        /// Activity source, null will be returned.
+        /// </summary>
+        /// <param name="name">The activity name.</param>
+        /// <returns>Activity or null.</returns>
+        private static Activity GetActivityMaybe(string name)
+        {
+            Activity act = null;
+            string enable = Environment.GetEnvironmentVariable(EnvironmentVariableSwitchName);
+            if (enable != null && (enable.Equals("true", StringComparison.OrdinalIgnoreCase) || enable.Equals("1")))
+                act = s_chatSource.CreateActivity(name, ActivityKind.Client);
+            return act;
+        }
+
+        /// <summary>
+        /// Set the tag on the activity, if the tag is present.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="name">The name of tag to set.</param>
+        /// <param name="value">Nullable value to be set.</param>
+        private void SetTagMaybe<T>(string name, Nullable<T> value) where T: struct
+        {
+            if (value.HasValue && m_activity != null)
+                m_activity.SetTag(name, value.Value);
+        }
+
+        /// <summary>
+        /// Return true if the activity for OpenTelemetry exists.
+        /// </summary>
+        public bool IsEnabled { get => m_activity != null; }
+
+        /// <summary>
+        /// Return true if the metrics are enabled.
+        /// </summary>
+        public bool AreMetricsOn { get => s_duration.Enabled && s_tokens.Enabled; }
+
+        /// <summary>
+        /// Create the instance of OpenTelemetryScope. This constructor logs request
+        /// and starts the execution timer.
+        /// </summary>
+        /// <param name="requestOptions">The request options used in the call.</param>
+        /// <param name="endpoint">The endpoint being called.</param>
+        public OpenTelemetryScope(ChatCompletionsOptions requestOptions, Uri endpoint)
+        {
+            m_recordedStreamingResponse = new();
+            m_activity = GetActivityMaybe($"Complete_{requestOptions.Model}");
```

Comment:
it seems `requestOptions.Model` is not always available in Azure AI Inference. We should make sure to set name to `chat` in this case

Reply:
Fixed.